pandas/core/dtypes/dtypes.py

```diff
@@ -894,13 +894,29 @@ def __new__(cls, subtype=None):
                 if m is not None:
                     subtype = m.group('subtype')
 
-            try:
-                subtype = pandas_dtype(subtype)
-            except TypeError:
-                raise TypeError("could not construct IntervalDtype")
+        # check subtype is numeric, datetime, or timedelta
+        valid_subtype = False
+
+        # test for number
+        try:
+            if np.issubdtype(subtype, np.number):
+                valid_subtype = True
+        except TypeError:
+            pass
+        # test for datetime
```

Comment:
this is not very idiomatic 

i would simply stick with pandas_dtype as it already does all the right things

Reply:
> i would simply stick with pandas_dtype

My experience here has been that using `pandas_dtype` is quite brittle within `IntervalDtype`.  Specifically you can easily get into a trap of circular dependencies, with `pandas_dtype` needing to register `IntervalDtype` as an extension dtype.  Also `pandas_dtype` is doing too much here, as it will parse invalid subtypes that we later need to exclude.  That being said, I think `pandas_dtype` could work here if we're careful and/or restructure some of the code.

 > this is not very idiomatic

I agree that the implementation here can be cleaned up.  I think using some of the functions in `pandas.core.dtypes.common` (e.g. `is_numeric_dtype`) would help clean things up and eliminate the need for the `try`/`except` blocks and the `valid_subtype` flag.  Does that seem acceptable?

Reply:
thank you for the feedback guys, it's been helpful so far!

I think I have a solution that sticks with using pandas_dtype but slightly modifies the IntervalDtype's construct_from_string method to avoid getting into the recursive trap. If it passes existing tests, it's probably the least intrusive fix possible.